sync-core/src/main/java/com/cloudant/sync/datastore/AttachmentStreamFactory.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2015 IBM Cloudant, Inc. All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.datastore;
+
+import com.cloudant.sync.datastore.encryption.EncryptedAttachmentInputStream;
+import com.cloudant.sync.datastore.encryption.EncryptedAttachmentOutputStream;
+import com.cloudant.sync.datastore.encryption.EncryptionKey;
+import com.cloudant.sync.datastore.encryption.KeyProvider;
+
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.SecureRandom;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+
+/**
+ * Class to manage returning appropriate streams to access on disk attachments.
+ *
+ * These may be encrypted and/or gzip encoded. This class handles returning an
+ * appropriate input/output stream chain to handle this.
+ *
+ * The input stream returned will allow reading unencrypted, unzipped data from
```

Comment:
This is a bit ambiguous - it would also be true to say it will allow reading encrypted, zipped data from the disk. It might be better to say something like *The input stream returned will contain the unencrypted, unzipped representation of the data from disk.*